showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseCodegenMetadataWriter.kt

```diff
@@ -31,7 +31,16 @@ internal class ShowkaseCodegenMetadataWriter(private val environment: XProcessin
         val autogenClass = TypeSpec.classBuilder(generatedClassName)
 
         showkaseMetadataSet.forEach { showkaseMetadata ->
-            val name = "${showkaseMetadata.showkaseGroup}_${showkaseMetadata.showkaseName}"
+
+            val name = if (
+                showkaseMetadata is ShowkaseMetadata.Component
+                && showkaseMetadata.componentIndex != null
+                && showkaseMetadata.componentIndex > 0
+            ) {
+                "${showkaseMetadata.showkaseGroup}_${showkaseMetadata.showkaseName}_${showkaseMetadata.componentIndex}"
```

Comment:
I know we were hoping for a better name but we can probably append `componentIndex` for now.